homeassistant/components/date/__init__.py

```diff
@@ -82,13 +82,13 @@ class DateEntity(Entity, cached_properties=CACHED_PROPERTIES_WITH_ATTR_):
     @final
     def device_class(self) -> None:
         """Return the device class for the entity."""
-        return None
+        return
```

Comment:
This is also wrong, I think. Properties should always return None explicitly since the return value will be consumed.

Reply:
From the runtime behavior there is no difference between `return` and `return None`. And as the function has a type annotation of `-> None`, there is nothing to be returned other than `None`.

```
>>> def a():
...     return
... 
>>> b = a()
>>> type(b)
<class 'NoneType'>
```

Reply:
Like I said, properties are always meant to have a return value for consumption and then we should `return None` explicitly, always.

Reply:
I have it now as a noqa.

Reply:
I agree with @MartinHjelmare about making it explicit.

But adding a niqab to perfectly valid code is not a good way to go! that signals the code is somehow not correct!

Reply:
There are four options in total:
- Adding the type hints as the property functions are in the parent classes, then there is not only None as return type
- Adding noqa directives
- Allowing the implicit None in all functions which only return None
- Ignoring the rule altogether

When none of the first three are good, then I would say that I put this rule into ignore and we skip it.

Reply:
Another option is to raise it with RUFF, and either change the default or add an option to explicitely exclude:
- properties
- pytest fixtures
- methods/functions that override parents with a not-None return type

Reply:
The question here is, is this something that just we want, or is this a common pattern in the Python environment? (I checked the original flake8-return, and this plugin does not care about these cases as well)

Reply:
Can you please keep `return None` on the properties?

Reply:
Ruff still doesn't like it. Either we change the declared return types to be the same as in the `Entity` class, or Ruff needs to be adjusted to take `cached_property` into account as well.